test/specs/dom/index.js

```diff
@@ -1,665 +1,699 @@
+                        );
+                    });
+                    it('Morphing to a URL should NOT issue a request after a canceled event', async () => {
+                        let originalRequestCount = fetchMock.calls().length;
+                        let firedEvent;
+
+                        const handler = async event => {
+                            firedEvent = event;
+                            event.preventDefault();
+
+                            window.removeEventListener(
+                                'palindrom-before-redirect',
+                                handler
+                            );
+                        };
+
+                        window.addEventListener(
+                            'palindrom-before-redirect',
+                            handler
+                        );
+
+                        await palindrom.morphUrl('/testURL');
+
+                        assert.equal(firedEvent.detail.href, '/testURL');
+                        
+                        await sleep();
+
+                        expect(originalRequestCount).to.equal(
+                            fetchMock.calls().length
+                        );
+                    });
+                });
+                describe('palindrom-after-redirect event', function() {
+                    beforeEach(async () => {
+                        // wait for Palindrom to call .listen (after finishing the ajax request)
+                        await sleep(300);
+                    });
+                    afterEach(fetchMock.reset);
+                    it('Morphing to a URL should dispatch the event after a successful request', async () => {
+                        const rel = getTestURL('newUrl', true);
+
+                        fetchMock.mock(rel, {
+                            status: 200,
+                            body: '{"hello": "world"}'
+                        });
+
+                        let firedEvent;
+
+                        const handler = event => {
+                            firedEvent = event;
+
+                            window.removeEventListener(
+                                'palindrom-after-redirect',
+                                handler
+                            );
+                        };
+
+                        window.addEventListener(
+                            'palindrom-after-redirect',
+                            handler
+                        );
+
+                        await palindrom.morphUrl(rel);
+                        await sleep()
+                        
+                        assert.equal(firedEvent.detail.href, rel);
+
+                        expect(window.location.pathname).to.equal(
+                            '/newUrl'
+                        );
+                    });
+
+                    it('Morphing to a URL should not dispatch the event after a failed request but should reject morphUrl call', async () => {
```

Comment:
```
- Morphing to a URL should throw an error...
+ Morphing to a URL should not dispatch the event...
```
This does not look liek something required in refactoring that changes errors and callback to events, https://github.com/Palindrom/Palindrom/issues/224

Reply:
My bad, it was not calling any callback nor event before. Now, it also rejects with same error. I'll update the message, and add assertion for Error and a message.